Running CLI command with "-U" results in a message containing `Did you mean "$0"`

## 🐛 Bug Report

Got this message when running CLI command with a bad parameter:

```
● Unrecognized CLI Parameter:
  Unrecognized option "U". Did you mean "$0"?
```

I seems to me that the message (`"$0"`) is wrong (looks like a misinterpreted regular expression replacement).

## To Reproduce

Launch jest cli command with `-U` instead of `-u`

## Expected behavior

I think the message is supposed to say:
`Unrecognized option "U". Did you mean "u"?`

## Link to repl or repo (highly encouraged)

N/A

## envinfo

full command and result

```
$ npm test -- -U

> <MyProject>@1.1.7 test <MyProjectPath>
> cross-env TZ=UTC NODE_ENV=test jest --ci "-U"

● Unrecognized CLI Parameter:

  Unrecognized option "U". Did you mean "$0"?

  CLI Options Documentation:
  https://jestjs.io/docs/en/cli.html

npm ERR! Test failed.  See above for more details.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 ### Fixes
 
+- `[jest-validate]` Show suggestion only when unrecognized cli param is longer than 1 character ([#10604](https://github.com/facebook/jest/pull/10604))
+
 ### Chore & Maintenance
 
 ### Performance

--- a/packages/jest-validate/src/__tests__/validateCLIOptions.test.js
+++ b/packages/jest-validate/src/__tests__/validateCLIOptions.test.js
@@ -41,3 +41,25 @@ test('fails for multiple unknown options', () => {
     validateCLIOptions(argv, options),
   ).toThrowErrorMatchingSnapshot();
 });
+
+test('does not show suggestion when unrecognized cli param length <= 1', () => {
+  const options = ['$0', '_', 'help', 'h'];
+  const argv = {
+    $0: true,
+    l: true,
+  };
+  expect(() =>
+    validateCLIOptions(argv, options),
+  ).toThrowErrorMatchingSnapshot();
+});
+
+test('shows suggestion when unrecognized cli param length > 1', () => {
+  const options = ['$0', '_', 'help', 'h'];
+  const argv = {
+    $0: true,
+    hell: true,
+  };
+  expect(() =>
+    validateCLIOptions(argv, options),
+  ).toThrowErrorMatchingSnapshot();
+});

--- a/packages/jest-validate/src/validateCLIOptions.ts
+++ b/packages/jest-validate/src/validateCLIOptions.ts
@@ -31,10 +31,10 @@ const createCLIValidationError = (
 
   if (unrecognizedOptions.length === 1) {
     const unrecognized = unrecognizedOptions[0];
-    const didYouMeanMessage = createDidYouMeanMessage(
-      unrecognized,
-      Array.from(allowedOptions),
-    );
+    const didYouMeanMessage =
+      unrecognized.length > 1
+        ? createDidYouMeanMessage(unrecognized, Array.from(allowedOptions))
+        : '';
     message =
       `  Unrecognized option ${chalk.bold(format(unrecognized))}.` +
       (didYouMeanMessage ? ` ${didYouMeanMessage}` : '');

